x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutoDetectResultProcessor.java

```diff
@@ -432,18 +432,18 @@ private void updateEstablishedModelMemoryOnJob() {
         // We need to make all results written up to and including these stats available for the established memory calculation
         persister.commitResultWrites(jobId);
 
+        try {
+            jobUpdateSemaphore.acquire();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOGGER.info("[{}] Interrupted acquiring update established model memory semaphore", jobId);
+            return;
+        }
+
         jobResultsProvider.getEstablishedMemoryUsage(jobId, latestBucketTimestamp, modelSizeStatsForCalc, establishedModelMemory -> {
             if (latestEstablishedModelMemory != establishedModelMemory) {
-
+                
                 client.threadPool().executor(MachineLearning.UTILITY_THREAD_POOL_NAME).submit(() -> {
```

Comment:
Can `submit()` throw? I'm not sure if it is necessary to wrap this in a try...catch and release the semaphore in the catch

Reply:
Also, it can use `execute()` rather than `submit()` as it's not interested in getting a `Future` to track completion.  Since you're off I'll push another commit that changes both things.